lib/models/cli-command-completer.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+var fs          = require('fs');
+var find        = require('lodash/collection/find');
+var omelette    = require('omelette');
+var Generator   = require('./cli-command-generator');
+
+/**
+  CliCommandCompleter is responsible for shell completion for partially
+  typed - and shell suggestion for fully typed cli commands.
+
+  @class CliCommandCompleter
+  @extends CoreObject
+  @constructor
+*/
+
+function CliCommandCompleter() {
+
+  try {
+    this.completion = omelette('ember');
+    this.template = this.loadTemplate('/node_modules/ember-cli/cli-commands.json');
+  } catch(e) {}
+
+}
+
+/**
+  Receives a relative path to the cli command cache file and returns its
+  content. If no cache file exists it will fallback to generate the JSON
+  on demand.
+
+  @private
+  @method loadTemplate
+ */
+
+CliCommandCompleter.prototype.loadTemplate = function(relPathToCache) {
+
+  var pathToCache = process.cwd() + relPathToCache;
+
+  if (fs.existsSync(pathToCache)) {
+
+    var content = fs.readFileSync(pathToCache, 'utf8');
+    return JSON.parse(content);
+
+  } else {
+
+    var generator = new Generator();
+    return generator.generateJSON();
+
+  }
+
+};
+
+/**
+  Initializes omelette completion and calls `this.handleInput` every
+  time a user hits `tab`.
+
+  @private
+  @method run
+ */
+
+CliCommandCompleter.prototype.run = function() {
+
+  this.completion.on('complete', this.handleInput.bind(this)).init();
+
+};
+
+/**
+  Generates an intent out of the user typed command line input and replies
+  with the correct results.
+
+  @private
+  @method handleInput
+  @param  {Number} fragment Fragment count (ignored)
+  @param  {String} word Parent token (ignored)
+  @param  {String} line Represents the command line input
+ */
+
+CliCommandCompleter.prototype.handleInput = function(fragment, word, line) {
+
+  var intent = {
+    arguments: this.getArgs(line),
+    isPremature: this.isPremature(line),
+    seeksOptions: this.seeksOptions(line)
+  };
+
+  try {
+    this.completion.reply(this.getResult(intent));
```

Comment:
When does this error? I suspect we don't actually want to swallow all exceptions this code-path could throw